src/gradualizer_prelude.erl

```diff
@@ -2,8 +2,70 @@
 
 %% This module contains specs to replace incorrect or inexact specs in OTP.
 
+-type ext_binary() :: binary().
+
+-spec erlang:apply(function(), [any()]) -> any().
+-spec erlang:apply(atom(), function(), [any()]) -> any().
+-spec erlang:binary_to_term(ext_binary()) -> any().
+-spec erlang:binary_to_term(ext_binary(), [safe | unsafe]) -> any() | {any(), pos_integer()}.
+-spec erlang:element(pos_integer(), tuple()) -> any().
+-spec erlang:erase() -> [{any(), any()}].
+-spec erlang:erase(any()) -> any() | undefined.
+-spec erlang:fun_info(function()) -> [{Item, any()}] when
+      Item :: arity
+	    | env
+	    | index
+	    | name
+	    | module
+	    | new_index
+	    | new_uniq
+	    | pid
+	    | type
+	    | uniq.
+-spec erlang:fun_info(function(), Item) -> {Item, any()} when
+      Item :: arity
+	    | env
+	    | index
+	    | name
+	    | module
+	    | new_index
+	    | new_uniq
+	    | pid
+	    | type
+	    | uniq.
+-spec erlang:get() -> [{any(), any()}].
+-spec erlang:get(any()) -> any() | undefined.
+-spec erlang:get_keys() -> [any()].
+-spec erlang:get_stacktrace() -> [{ module()
+				  , atom()
+				  , arity() | [any()]
+				  , [{file, string()}|{line, pos_integer()}]}].
 -spec erlang:hd([A, ...]) -> A.
+-spec erlang:max(any(), any()) -> any().
+-spec erlang:min(any(), any()) -> any().
+-spec erlang:port_call(port() | atom(), integer(), any()) -> any().
+%% TODO: erlang:process_info
+-spec erlang:put(any(), any()) -> any().
+-spec erlang:raise(Class, Reason, Stacktrace) -> no_return() when
+      Class      :: error | exit | throw,
+      Reason     :: any(),
+      Stacktrace :: [{module(), atom(), arity() | [any()]} |
+		     {function(), [any()]}] |
+		    [{module(), atom(), arity() | [any()], [{atom(), any()}]} |
+		     {function(), [any()], [{atom(), any()}]}].
+-spec erlang:send(pid() | port() | atom() | {atom(), node()}, any()) -> any().
+-spec erlang:send(pid() | port() | atom() | {atom(), node()}, any()
+		 , [nosuspend | noconnect])
+		 -> ok | nosuspend | noconnect.
+%% TODO: erlang:system_info({allocator, atom()) -> [term()]
+%% TODO: erlang:system_info({allocator_sizes, atom()) -> [term()]
+%% TODO: erlang:system_info(os_monotonic_time_source) -> [{atom(), term()}].
+%% TODO: erlang:system_info(os_system_time_source) -> [{atom(), term()}].
+%% TODO: erlang:system_info(Item :: c_compiler_used) -> {atom(), term()}.
+%% TODO: erlang:system_info(Item :: check_io) -> [term()]
```

Comment:
Intersection type for this one? If it's incomplete, just add a catch-any clause `(any()) -> any()` in the end.

Reply:
Yes, it's defined as a massive intersection type in otp right now. I've listed the types which I found that were problematic but there are many, many more cases. I'm just going to leave this function for another diff.